webapp/channels/src/components/permissions_gates/any_team_permission_gate/any_team_permission_gate.tsx

```diff
@@ -18,23 +17,18 @@ export type Props = {
     /**
      * Invert the permission (used for else)
      */
-    invert: boolean;
+    invert?: boolean;
 
     /**
      * Content protected by the permissions gate
      */
     children: React.ReactNode;
-}
+};
 
-export default class AnyTeamPermissionGate extends React.PureComponent<Props> {
-    static defaultProps = {
-        invert: false,
-    };
+const AnyTeamPermissionGate: React.FC<Props> = ({
+    hasPermission,
+    invert = false,
+    children,
+}) => <>{hasPermission !== invert ? children : null}</>
```

Comment:
I appreciate the simplicity, but I feel having this in multiple lines is more readable. Something like this:
```suggestion
}) => (
    <>
        {hasPermission !== invert ? children : null}
    </>
);
```